dao/src/main/java/org/thingsboard/server/dao/entity/BaseEntityService.java

```diff
@@ -79,7 +85,51 @@ public PageData<EntityData> findEntityDataByQuery(TenantId tenantId, CustomerId
         validateId(tenantId, INCORRECT_TENANT_ID + tenantId);
         validateId(customerId, INCORRECT_CUSTOMER_ID + customerId);
         validateEntityDataQuery(query);
-        return this.entityQueryDao.findEntityDataByQuery(tenantId, customerId, query);
+
+        if (EntityFilterType.RELATIONS_QUERY.equals(query.getEntityFilter().getType()) || query.getPageLink().getTextSearch() != null) {
```

Comment:
`query.getPageLink().getTextSearch() != null` - text search better to check with no blank value to do execute query in case empty string (not null) was provided